src/langauge-servers/configStore.ts

```diff
@@ -181,31 +187,36 @@ class LangServerConfigStore {
         }
     }
 
+    private installPythonLS() {
+        const {stderr} = cp.spawnSync(PYTHON_EXEC_FILENAME,
+            ["-m", "pip", "install", "python-lsp-server[all]"],
+            { cwd: this.serverLocations["python"], encoding: "utf-8" }
+        );
+        if (stderr) {
+            throw new Error(stderr);
+        }
+    }
+
     private installPythonLanguageServerIfNeeded() {
         if (!this.serverLocations["python"]) {
             this.findPythonLocation();
         }
         try {
-            const stdout = cp.execFileSync(
-                PYTHON_FILENAME,
+            const { stdout, stderr } = cp.spawnSync(
+                PYTHON_EXEC_FILENAME,
                 ["-m", "pip", "show", "python-lsp-server"],
                 { cwd: this.serverLocations["python"], encoding: "utf-8" }
             );
             const trimmedOutput = _.trim(stdout);
-            if (trimmedOutput.startsWith("WARNING: Package(s) not found:")) {
-                cp.execFileSync(PYTHON_FILENAME,
-                    ["-m", "pip", "install", "python-lsp-server[all]"],
-                    { cwd: this.serverLocations["python"], encoding: "utf-8" }
-                );
+            const trimmedError = _.trim(stderr);
```

Comment:
if stderr is not one of the two you are looking for you need to let the user know somehow. Re-throw maybe?